From 2914df5ff196f28524492b9021bc50298d636f71 Mon Sep 17 00:00:00 2001
From: Andreas Schmidt <andreas.schmidt@wago.com>
Date: Fri, 25 Jan 2019 09:53:18 +0100
Subject: [PATCH] dts: 8211: fix indets and sync mv88e6321 to 8215 dts

PFC200 8211 use the same ethnet-switch (mv88e6321) as
PFC200 8215. This patch synchronize the mv88e6321 part,
exept the port 0 and 1. PFC200 8211 has on port 0 and 1
a SFP interface.

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 arch/arm/boot/dts/am335x-pfc-750_8211.dts | 259 +++++++++++++++++++-----------
 1 file changed, 161 insertions(+), 98 deletions(-)

diff --git a/arch/arm/boot/dts/am335x-pfc-750_8211.dts b/arch/arm/boot/dts/am335x-pfc-750_8211.dts
index c581abe..8c038cb 100644
--- a/arch/arm/boot/dts/am335x-pfc-750_8211.dts
+++ b/arch/arm/boot/dts/am335x-pfc-750_8211.dts
@@ -9,46 +9,29 @@
 
 #include "am335x-pfc-750_821x.dtsi"
 #include "devconf.dtsi"
+#include "am335x-pfc-750_8208-dcan.dtsi"
 
 /{
 	model = "WAGO PFC200 750-8211";
 	compatible = "wago,am335x-pfc-750_821x-1004", "wago,am335x-pfc", "ti,am33xx";
 
-        swcfg_mv88e6321 {
-                compatible = "swcfg,mv88e6321";
-                status = "okay";
+	swcfg_mv88e6321 {
+		compatible = "swcfg,mv88e6321";
+		status = "okay";
 
-                swcfg,mii-bus = <&davinci_mdio>;
+		swcfg,mii-bus = <&davinci_mdio>;
 
-                swcfg,alias = "mv88e6321";
-                swcfg,cpu_port = <5>;
-                swcfg,ports = <7>;
-                swcfg,vlans = <1>;
-        };
-};
-
-&ksz8863 {
-	status = "disabled";
-};
-
-&swcfg_ksz8863 {
-	status = "disabled";
-};
-
-&wsysinit {
-	tty,rs232-485 = "nop";
-	profinet-capable;
-};
-
-&dsa {
-	status = "disabled";
-};
-
-&bitbang_mdio0 {
-	status = "disabled";
+		swcfg,alias = "mv88e6321";
+		swcfg,cpu_port = <5>;
+		swcfg,ports = <7>;
+		swcfg,vlans = <1>;
+	};
 };
 
 &am33xx_pinmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <&ethernet_irq_pins>;
+
 	davinci_mdio_default_pins: pinmux_davinci_mdio_default_pins {
 		pinctrl-single,pins = <
 			/* MDIO */
@@ -64,6 +47,39 @@
 			0x14c (PIN_INPUT_PULLDOWN | MUX_MODE7)
 		>;
 	};
+
+	ethernet_irq_pins: pinmux_ethernet_irq_pins {
+		pinctrl-single,pins = <
+			0x0dc (PIN_INPUT | MUX_MODE7) /* lcd_data15.gpio0_11 */ //MV88E6321-nINT
+		>;
+	};
+};
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart1_pins>;
+	status = "okay";
+};
+
+&cppi41dma {
+	status = "okay";
+};
+
+&ksz8863 {
+	status = "disabled";
+};
+
+&wsysinit {
+	tty,rs232-485 = "nop";
+	profinet-capable;
+};
+
+&dsa {
+	status = "disabled";
+};
+
+&bitbang_mdio0 {
+	status = "disabled";
 };
 
 &davinci_mdio {
@@ -72,82 +88,129 @@
 	pinctrl-1 = <&davinci_mdio_sleep_pins>;
 	status = "okay";
 
-        phy3: ethernet-phy@3 {
-                compatible = "ethernet-phy-ieee802.3-c22";
-                reg = <3>;
-        };
-
-        phy4: ethernet-phy@4 {
-                compatible = "ethernet-phy-ieee802.3-c22";
-                reg = <4>;
-        };
-
-        switch: switch@0 {
-                compatible = "marvell,mv88e6085";
-                #address-cells = <1>;
-                #size-cells = <0>;
-                reg = <6>;
-                dsa,member = <0 0>;
-                reset-gpios = <&io_expander_70 4 GPIO_ACTIVE_LOW>;
-                phy-scan;
-
-                ports {
-                        #address-cells = <1>;
-                        #size-cells = <0>;
-
-                        port@0 {
-                                reg = <0>;
-                                label = "ethX1";
-                                phy-mode = "sgmii";
+	switch: switch@0 {
+		compatible = "marvell,mv88e6085";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+		dsa,member = <0 0>;
+		reset-gpios = <&io_expander_70 4 GPIO_ACTIVE_LOW>;
+		phy-scan;
+		eeprom-length = <256>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <11 IRQ_TYPE_LEVEL_LOW>;
+		interrupt-controller;
+		#interrupt-cells = <2>;
+		dsa,enable-on-boot;
+
+		mdio {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			int_phy3: ethernet-phy@3 {
+				compatible = "ethernet-phy-id0141.0C00", "ethernet-phy-ieee802.3-c22";
+				reg = <3>;
+				max-speed = <100>;
+				interrupt-parent = <&switch>;
+				interrupts = <3 IRQ_TYPE_LEVEL_HIGH>;
+				eee-broken-100tx;
+				eee-broken-1000t;
+			};
+
+			int_phy4: ethernet-phy@4 {
+				compatible = "ethernet-phy-id0141.0C00", "ethernet-phy-ieee802.3-c22";
+				reg = <4>;
+				max-speed = <100>;
+				interrupt-parent = <&switch>;
+				interrupts = <4 IRQ_TYPE_LEVEL_HIGH>;
+				eee-broken-100tx;
+				eee-broken-1000t;
+			};
+		};
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				label = "ethX1";
+				phy-mode = "sgmii";
+				fixed-link {
+					speed = <100>;
+					full-duplex;
+				};
+			};
+
+			port@1 {
+				reg = <1>;
+				label = "ethX2";
+				phy-mode = "sgmii";
 				fixed-link {
 					speed = <100>;
 					full-duplex;
 				};
-                        };
+			};
+
+			/*port@2 {
+				reg = <2>;
+				label = "cpu";
+				ethernet = <&mac>;
+				fixed-link {
+					speed = <100>;
+					full-duplex;
+				};
+			};*/
+
+			port@3 {
+				reg = <3>;
+				label = "ethX11";
+				phy-handle = <&int_phy3>;
+				phy-mode = "rmii";
+			};
+
+			port@4 {
+				reg = <4>;
+				label = "ethX12";
+				phy-handle = <&int_phy4>;
+				phy-mode = "rmii";
+			};
 
-                        port@1 {
-                                reg = <1>;
-                                label = "ethX2";
-                                phy-mode = "sgmii";
-       				fixed-link {
+			port@6 {
+				reg = <6>;
+				label = "cpu";
+				ethernet = <&mac>;
+				fixed-link {
 					speed = <100>;
 					full-duplex;
 				};
 			};
+		};
+	};
+};
+
+&led_bar0 {
+	bf_r: bf-red@0 {
+		label = "bf-red";
+		reg = <0>;
+		linux,default-trigger = "none";
+	};
 
-                        /*port@2 {
-                                reg = <2>;
-                                label = "cpu";
-                                ethernet = <&mac>;
-                                fixed-link {
-                                        speed = <100>;
-                                        full-duplex;
-                                };
-                        };*/
-
-                        port@3 {
-                                reg = <3>;
-                                label = "ethX11";
-                                phy-handle = <&phy3>;
-                                phy-mode = "rmii";
-                        };
-
-                        port@4 {
-                                reg = <4>;
-                                label = "ethX12";
-                                phy-handle = <&phy4>;
-                                phy-mode = "rmii";
-                        };
-
-                        port@6 {
-                                reg = <6>;
-                                label = "cpu";
-                                ethernet = <&mac>;
-                                fixed-link {
-                                        speed = <100>;
-                                        full-duplex;
-                                };
-                        };
-                };
-        };
+	bf_g: bf-green@1 {
+		label = "bf-green";
+		reg = <1>;
+		linux,default-trigger = "none";
+	};
+
+	dia_r: dia-red@2 {
+		label = "dia-red";
+		reg = <2>;
+		linux,default-trigger = "none";
+	};
+
+	dia_g: dia-green@3 {
+		label = "dia-green";
+		reg = <3>;
+		linux,default-trigger = "none";
+	};
 };
-- 
2.7.4

